packages/block-editor/src/hooks/style.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import { assign, has, mapKeys, kebabCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { hasBlockSupport } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { COLOR_SUPPORT_KEY } from './color';
+
+const styleSupportKeys = [ COLOR_SUPPORT_KEY ];
```

Comment:
This is a generic hook that just applies the automatic per block, global CSS variables generation if custom colors are used.

The line above is the only line that should be updated if we add new "support flags" for other UIs (lineHeight...)

Reply:
Could we unconditionally add the hooks without checking hasStyleSupport?

Reply:
we could but it could break existing blocks that have an attribute named "style"

Reply:
In that case could instead we just check if the block has an explicitly defined style attribute and if yes we disable this mechanism?

Reply:
That's an option, there's still a small performance overhead in running these hooks for all the blocks. Let's keep it as is for now and reconsider as we add more support flags